test/cmocka/src/lib/lib/memset.c

```diff
@@ -0,0 +1,100 @@
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Michal Jerzy Wierzbicki <michalx.wierzbicki@linux.intel.com>
+ */
+#include <sof/alloc.h>
+
+#include <stdarg.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+static void test_lib_lib_memset_char_change_none(void **state)
+{
+	const char *before = "aaa";
+	char str[]  = "aaa";
+	char *p;
+
+	(void) state;
+	assert_int_equal(rstrcmp(before, str), 0);
+
+	p = memset(str, 'b', 0);
+	assert_int_equal(rstrcmp(before, str), 0);
+	assert_int_equal(rstrcmp(before, p), 0);
+}
+
+static void test_lib_lib_memset_char_change_first(void **state)
+{
+	const char *before = "aaa";
+	const char *after  = "baa";
+	char str[] = "aaa";
+	char *p;
+
+	(void) state;
+	assert_int_equal(rstrcmp(before, str), 0);
+	assert_int_not_equal(rstrcmp(after, str), 0);
+
+	p = memset(str, 'b', 1);
+	assert_int_equal(rstrcmp(after, str), 0);
+	assert_int_not_equal(rstrcmp(before, str), 0);
+
+	assert_int_equal(rstrcmp(after, p), 0);
+	assert_int_not_equal(rstrcmp(before, p), 0);
+}
+
+static void test_lib_lib_memset_char_change_middle(void **state)
+{
+	const char *before = "aaa";
+	const char *after  = "aba";
+	const char *afterP = "ba";
+	char str[]  = "aaa";
+	char *p;
+
+	(void) state;
+	assert_int_equal(rstrcmp(before, str), 0);
+	assert_int_not_equal(rstrcmp(after, str), 0);
+
+	p = memset(str+1, 'b', 1);
+	assert_int_equal(rstrcmp(after, str), 0);
+	assert_int_not_equal(rstrcmp(before, str), 0);
+
+	assert_int_equal(rstrcmp(afterP, p), 0);
+	assert_int_not_equal(rstrcmp(before, p), 0);
```

Comment:
Please, split functions to have 1 assert pert ut.

Reply:
@jajanusz Done, see https://github.com/thesofproject/sof/pull/202#issuecomment-412501550